### ISSUE TITLE
issue with tcmalloc ?

### ISSUE THREAD

**Issue by FlorianIragne** [facebookresearch/faiss#3212]
# Summary
Hi, it seems that faiss isn't playing well with tcmalloc. I have a 20GB faiss index, which is taking roughly 20/21GB once loaded within my app. I haven't been able to catch the triggering event, but after a few hours, the memory usage is climbing up to 40, then 60/80+ GB. If i use my own index instead of faiss, there's no such issue.
I've run valgrind for leak check and found none. I'm also running with OMP_NUM_THREADS=1, since my application handles the threading (and i'm using locks where appropriate with faiss index).

If i link to standard glibc, there's no such behavior. 

I've searched for know issues, or anything related but didn't find a clue.

# Platform

Debian bookworm, intel or amd cpus (24 cores / 48 threads, AVX2)
Faiss version: latest commit, 7f3e0a349f6ff5752ff3e36ec0b2e03bea9007a0
libtcmalloc-minimal4 v2.10-1 from debian bookworm repo

Installed from: compiled by myself
Faiss compilation options: cmake -DFAISS_ENABLE_PYTHON=OFF -DFAISS_ENABLE_GPU=OFF -DBUILD_TESTING=OFF -DBUILD_SHARED_LIBS=ON -DFAISS_OPT_LEVEL=avx2 -DCMAKE_BUILD_TYPE=Release -DBLA_VENDOR=Intel10_64_dyn  -DCMAKE_INSTALL_PREFIX=/usr 

Running on:
- [ x] CPU
- [ ] GPU

Interface: 
- [ x] C++
- [ ] Python

# Reproduction instructions
Index is IndexIVFQP with IndexHNSWFlat as quantizer

**Comment by mdouze** [comment]
Interesting, unfortunately, I am not aware of this, and I don't know what additional debugging steps you could take. Maybe try to disable openmp alogether (ie. remove -fopenmp and ignore the openmp pragmas) ?

**Comment by FlorianIragne** [comment]
Hi, Thanks for your answer. I've already tried commenting the pragmas, but did not removed -fopenmp. I was a bit reluctant to do the last one since there's a bunch of omp_get_thread_num/num_threads throughout the code, but it should work. 
Anyway, there was no difference in RAM usage behavior. 
What's weirder is that i've tried to compile a fresh checkout of the git, with openmp, same compilation options, etc, and it's working now. A diff between my old checkout and the new one doesn't show any difference. I've already checked that there was no legacy/trace of old faiss builds on the vms running the code (could explain the weird issue), and there were none. 

I'm closing this right now as it is working fine for a few days with tcmalloc, but i'll try to investigate more should it shows again
Thanks !
